pkg_pipeline/single_build.R

```diff
@@ -0,0 +1,167 @@
+#! /usr/local/bin/Rscript
+
+success <- c(
+    check = FALSE,
+    test = FALSE,
+    lint = FALSE,
+    valdoc = FALSE
+)
+
+# Getting package information ----
+pkg_name <- read.dcf("DESCRIPTION")[1, "Package"]
+pkg_version <- read.dcf("DESCRIPTION")[1, "Version"]
+
+# Building ----
+
+message("############################")
+message("###### INSTALLING (S) ######")
+message("############################")
+
+pak::local_install_deps(".", upgrade=FALSE, ask=FALSE, dependencies = TRUE)
+devtools::install(upgrade = FALSE, args = "--install-tests")
+
+message("############################")
+message("###### INSTALLING (F) ######")
+message("############################")
+
+# Checking ----
+
+message("##########################")
+message("###### CHECKING (S) ######")
+message("##########################")
+
+check_results <- rcmdcheck::rcmdcheck(error_on = "never", args = "--no-tests")
```

Comment:
would definitely not serialize steps in a big file. KISS, remember?

Reply:
+ why wait for fails if you can sees them all at once and hopefully fix them in one go?

Reply:
I would argue that it is simpler to maintain a single R file that runs all the steps and offers a final summary with all the errors found. You don't wait more than having separate pods or have less information from my point of view, a table is included in the Action Summary, and in the end you have to check the logs anyway to find the specific errors.

Reply:
old wheel to a new cart. disagree

Reply:
@sorinvoicu
Strongly agree with @zsigmas on this one and I'm ready to die on this hill.

"One cart carrying all the things we need VS several carts that need coordination."

Reply:
be my guest

Reply:
I have slept on the matter and have a few comments.

My combative wording ("ready to die on this hill") stems from your unfortunate "KISS, remember?" except I'm not insulting (accidentally or purposefully) anyone. In retrospect, I could have been more civil.

On the topic of simplicity, yes, I do think one big file that executes top-to-bottom is simpler than several files strung together through github actions. It is simpler because (in subjective decreasing order of importance):
1. You only need to look in a single place to understand how the build process works.
2. Related to the previous point, all build information is available at the same time on a single pod, making its manipulation trivial.
3. You can run it locally without having to mirror github machinery on your machine (a luxury that our internal colleagues don't have). This means you can tweak it locally and have a pretty good sense that it's going to work without having to go through minutes-long trial and error, which not only makes us miserable but costs the company money.
4. We have less work to do whenever github decides to deprecate features we rely on (https://github.blog/changelog/label/actions+deprecation/)
5. It makes it easier for potential users to take our code and use it mostly as-is outside of github. Our choice of hosting platform should not dictate theirs.
6. It is more environmentally friendly, because we save the cost of spinning up several containers (drop in the ocean, I know, considering we're encouraging the use of R).

I have numbered them so that you can comment on any of them individually and see where the nature of our disagreement lies. I would also like to know your reasons for thinking that a split `.R` file is simpler.

On the topic of my username: I revised it from `mlebs` to the obviously superior `ml-ebs-ext`, that make it clear that I'm an external working through EBS and my initials are "m" and "l".

Reply:
@sorinvoicu Sorry, I failed to mention your nickname yesterday. This is an important issue because it sits upstream of developers and dictates the way they work. I would like to hear your reasons for thinking that a split `.R` file is simpler. Even if we fail to convince each other, the whole team can benefit from this conversation.